### PR TITLE
[Priest] Cooldown reduction cleanup

### PR DESCRIPTION
**Pull request by taherbert**
* Adds void summoner for discipline
* Remove shadow spec conditional for manipulation (mindgames)
* move train of thought to constructor initialization for consistency

### CODE REVIEW

File: engine/class_modules/priest/sc_priest.cpp [hunk 1]
```diff
@@ -559,11 +570,20 @@ struct smite_t final : public priest_spell_t
   {
     priest_spell_t::execute();
 
-    if ( priest().talents.manipulation.enabled() &&
-         ( priest().specialization() == PRIEST_HOLY || priest().specialization() == PRIEST_DISCIPLINE ) )
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.cpp]: This is actually not true based on the tooltip 
image

File: engine/class_modules/priest/sc_priest.hpp [hunk 2]
```diff
@@ -773,9 +775,10 @@ struct priest_action_t : public Base, public parse_buff_effects_t<priest_td_t>
     parse_buff_effects( p().buffs.dark_evangelism, p().talents.shadow.dark_evangelism );
     parse_buff_effects( p().buffs.surge_of_darkness, false );  // Mind Spike instant cast
     if ( p().is_ptr() )
-        parse_buff_effects( p().buffs.mind_melt, p().talents.shadow.mind_melt );                 // Mind Blast instant cast and Crit increase
+      parse_buff_effects( p().buffs.mind_melt,
+                          p().talents.shadow.mind_melt );  // Mind Blast instant cast and Crit increase
     else
-        parse_buff_effects( p().buffs.mind_melt );  // Mind Blast instant cast and Crit increase
+      parse_buff_effects( p().buffs.mind_melt );  // Mind Blast instant cast and Crit increase
```
Comment (seanpeters86) [inline engine/class_modules/priest/sc_priest.hpp]: this entire indent change is very scary, could you just wrap it in brackets or just not touch it?

**Review by seanpeters86 — CHANGES_REQUESTED** [review]

File: engine/class_modules/priest/sc_priest.cpp (hunk 1 above)
Comment (taherbert) [reply engine/class_modules/priest/sc_priest.cpp]: Added these safety checks back in here, as well as for mind blast: https://github.com/simulationcraft/simc/pull/7413/files/3dfbcd0b7ae91b4468f581ad406ba05348644fda#diff-3642d3768d1526e985a4c06a468c29788411f96429999aa347a669aa7c32d9adL90

File: engine/class_modules/priest/sc_priest.hpp (hunk 2 above)
Comment (taherbert) [reply engine/class_modules/priest/sc_priest.hpp]: wrapped in brackets

**Review by seanpeters86 — APPROVED** [review]

**Pull request merged**